### PR TITLE
Issue #1052: Document theme_debug.

### PR DESCRIPTION
**Pull request by findlabnet**
Implementation for https://github.com/backdrop/backdrop-issues/issues/1052

### CODE REVIEW

File: core/modules/system/system.install [hunk 1]
```diff
@@ -568,7 +568,20 @@ function system_requirements($phase) {
...
+  
+  if ($phase == 'runtime') {
+    // Inform user if theme_debug is enabled.
+    $theme_debug_enabled = config_get('system.core', 'theme_debug');
+    if ($theme_debug_enabled) {
+      $requirements['theme_debug'] = array(
+        'title' => $t('Theme debug'),
+        'value' => $t('Enabled'),
+        'severity' => REQUIREMENT_WARNING,
+        'description' => $t('Theme debug is enabled, probably via "Devel" module. Do not forget to disable this module when you have finished testing and this site goes live.'),
```
Comment (jenlampton) [inline core/modules/system/system.install]: > Theme debug is enabled, probably via "Devel" module. 

I'm not sure I like this language. "Theme debug" is a strange noun to put in the user interface. Let's say "Theme debugging" here instead. The "probably via" also doesn't really fit with our style.  

> Do not forget to disable this module when you have finished testing and this site goes live.

Disabling the devel module will not actually change the theme debug setting. We should be more specific about what people should do to solve the problem that caused the warning. Also, this setting is more for development than for testing. hm..

How about something like:

`Theme debugging is currently enabled, and should be disabled on a live site. This setting can be changed using the devel module.`

File: core/modules/system/system.admin.inc [hunk 2]
```diff
@@ -400,6 +407,12 @@ function system_theme_settings($form, &$form_state, $key) {
     '#type' => 'submit',
     '#value' => t('Save theme settings'),
   );
+  
+  // Inform user if theme_debug is enabled.
+  $theme_debug_enabled = config_get('system.core', 'theme_debug');
+  if ($theme_debug_enabled) {
+    backdrop_set_message(t('Theme debug is enabled. See the <a href="@theme_debug">Status report</a> page for more info.', array('@theme_debug' => base_path() . 'admin/reports/status')), 'warning');
```
Comment (jenlampton) [inline core/modules/system/system.admin.inc]: `Theme debugging`

File: core/modules/system/system.admin.inc [hunk 3]
```diff
@@ -217,6 +217,13 @@ function system_themes_page() {
   backdrop_alter('system_themes_page', $theme_groups);
 
   $admin_form = backdrop_get_form('system_themes_admin_form', $admin_theme_options);
+
+  // Inform user if theme_debug is enabled.
+  $theme_debug_enabled = config_get('system.core', 'theme_debug');
+  if ($theme_debug_enabled) {
+    backdrop_set_message(t('Theme debug is enabled. See the <a href="@theme_debug">Status report</a> page for more info.', array('@theme_debug' => base_path() . 'admin/reports/status')), 'warning');
```
Comment (jenlampton) [inline core/modules/system/system.admin.inc]: `Theme debugging`

File: core/modules/installer/installer.manager.inc [hunk 4]
```diff
@@ -277,6 +277,12 @@ function installer_manager_update_form($form, $form_state = array(), $context) {
       '#weight' => 120,
     );
   }
+  
+  // Inform user if theme_debug is enabled.
+  $theme_debug_enabled = config_get('system.core', 'theme_debug');
+  if ($theme_debug_enabled && $context == 'theme') {
+    backdrop_set_message(t('Theme debug is enabled. See the <a href="@theme_debug">Status report</a> page for more info.', array('@theme_debug' => base_path() . 'admin/reports/status')), 'warning');
```
Comment (jenlampton) [inline core/modules/installer/installer.manager.inc]: `Theme debugging`

File: core/modules/installer/installer.pages.inc [hunk 5]
```diff
@@ -844,6 +844,12 @@ function installer_browser_filters_form($form, &$form_state, $type) {
       '#value' => t('Clear'),
     );
   }
+  
+  // Inform user if theme_debug is enabled.
+  $theme_debug_enabled = config_get('system.core', 'theme_debug');
+  if ($theme_debug_enabled && $type == 'theme') {
+    backdrop_set_message(t('Theme debug is enabled. See the <a href="@theme_debug">Status report</a> page for more info.', array('@theme_debug' => base_path() . 'admin/reports/status')), 'warning');
```
Comment (jenlampton) [inline core/modules/installer/installer.pages.inc]: `Theme debugging`

**Comment by findlabnet** [comment]
Thanks for this review @jenlampton . You right, language construction should be changed.

> Disabling the devel module will not actually change the theme debug setting. 

I did it in conjunction with changes in "devel" module (https://github.com/backdrop-contrib/devel/pull/28/files) - see changed devel.install. If someone disable module without disabling "debugging mode" - this will actually disable the theme debug setting.

**Comment by findlabnet** [comment]
> I'm not sure I like this language.

Language fixed as recommended. @jenlampton, thanks again.

**Comment by quicksketch** [comment]
Thanks @findlabnet! I'm working on polishing this up for merging. I added a link to the devel module on the status report page, and removed the warnings from Installer module. Now the warnings are only present on the theme listing and theme settings pages (and the status report of course).

New PR at https://github.com/backdrop/backdrop/pull/1699.

**Pull request closed without merging**